api/server/server.go

```diff
@@ -1640,8 +1640,9 @@ func allocateDaemonPort(addr string) error {
 		return fmt.Errorf("failed to lookup %s address in host specification", host)
 	}
 
+	pa := portallocator.Get()
```

Comment:
It might be better to follow the convention of the [flag](http://godoc.org/flag) package. In addition to having an "instance" creation function, like `NewAllocator`, promote all the same methods to the package level. For the instances that can't be created with `init`, such as those that do file io, call `sync.Do` at the start of each method to ensure it is created. It is a little more code on the package side by makes the pattern clear.

Let me know if I am not clear above.